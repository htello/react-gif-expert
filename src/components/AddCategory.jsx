import { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (ev) => {
    setInputValue(ev.target.value);
  };

  const onSubmit = (ev) => {
    // console.log("mensaje desde el submit");
    ev.preventDefault();

    if (inputValue.trim().length <= 1) return;
    setInputValue("");
    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        onChange={onInputChange}
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
