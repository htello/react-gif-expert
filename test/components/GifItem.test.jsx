import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("pruebas en <GifItem>", () => {
  const title = "este es el título";
  const url =
    "https://static.wikia.nocookie.net/marvelcomicsenciclopedia/images/c/c4/La_cosa.jpg/revision/latest?cb=20120605191426";
  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("debe mostrar la imagen con el url indicado y el alt indicado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(src);
    expect(alt).toBe(alt);
  });

  test("debe mostrar el título en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
