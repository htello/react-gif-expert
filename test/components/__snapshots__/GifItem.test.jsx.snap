// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="este es el título"
      src="https://static.wikia.nocookie.net/marvelcomicsenciclopedia/images/c/c4/La_cosa.jpg/revision/latest?cb=20120605191426"
    />
    <p>
      este es el título
    </p>
  </div>
</div>
`;
